qt/schedulewidget.py

```diff
@@ -0,0 +1,289 @@
+# SPDX-FileCopyrightText: © 2008-2022 Oprea Dan
+# SPDX-FileCopyrightText: © 2008-2022 Bart de Koning
+# SPDX-FileCopyrightText: © 2008-2022 Richard Bailey
+# SPDX-FileCopyrightText: © 2008-2022 Germar Reitze
+# SPDX-FileCopyrightText: © 2008-2022 Taylor Raak
+# SPDX-FileCopyrightText: © 2024 Christian BUHTZ <c.buhtz@posteo.jp>
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In time" which is released under GNU
+# General Public License v2 (GPLv2).
+# See file LICENSE or go to <https://www.gnu.org/licenses/#GPL>.
+import datetime
+from PyQt6.QtWidgets import (QHBoxLayout,
+                             QFormLayout,
+                             QGroupBox,
+                             QWidget,
+                             QLabel,
+                             QLineEdit,
+                             QSpinBox,
+                             QCheckBox)
+import config
+import tools
+import qttools
+import combobox
+
+
+class ScheduleWidget(QGroupBox):
+    """Widget about schedule snapshots.
+
+    That widget is used in the 'General' tab of the 'Manage profiles' dialog.
+    """
+    def __init__(self, parent):
+        super().__init__(title=_('Schedule'), parent=parent)
+
+        main_layout = QFormLayout(self)
+
+        def _create_form_entry(label: str, widget: QWidget=None) -> int:
+            """Helper to create a row with a label and widget in the form
+            layout.
+
+            The returned index is used later to toggle visibility of that
+            row.
+            """
+            if widget:
+                main_layout.addRow(label, widget)
+            else:
+                lbl = QLabel(label)
+                lbl.setWordWrap(True)
+                main_layout.addRow(lbl)
+
+            return main_layout.rowCount() - 1
+
+        # Schedule modes
+        self._combo_schedule_mode = self._schedule_mode_combobox()
+        main_layout.addRow(self._combo_schedule_mode)
+
+        # Day
+        self._combo_day = self._day_combobox()
+        self._rowidx_day = _create_form_entry(_('Day:'), self._combo_day)
+
+        # Weekday
+        self._combo_weekday = self._weekday_combobox()
+        self._rowidx_weekday = _create_form_entry(
+            _('Weekday:'), self._combo_weekday)
+
+        # Time (formerly known as Hour)
+        self._combo_time = self._time_combobox()
+        self._rowidx_time = _create_form_entry(
+            _('Time:'), self._combo_time)
```

Comment:
Modified the string `Hour:` into `Time:` because it fits IMHO better to values like `['0:00', '01:00', ..., '23:00']`.